R/inventory_table.R

```diff
@@ -0,0 +1,75 @@
+  #checkmate::assert_string(date)
+  #checkmate::assert_date(data[[date]])
+  #checkmate::assert_character(elements)
+  #assert_column_names(data, elements)
+  #if (!is.null(station)) assert_column_names(data, station)
+  #if (!is.null(year)) assert_column_names(data, year)
+  #if (!is.null(month)) assert_column_names(data, month)
+  #if (!is.null(day)) assert_column_names(data, day)
+  
+  if(is.null(year)) {
+    year <- ".year"
+    data[[year]] <- lubridate::year(data[[date]])
+  }
+  if(is.null(month)) {
+    month <- ".month"
+    data[[month]] <- lubridate::month(data[[date]])
+  }
+  if(is.null(day)) {
+    day <- ".day"
+    data[[day]] <- lubridate::day(data[[date]])
+  }
+  
+  if(is.null(station)){
+    inventory_data <- data %>%
+      tidyr::pivot_longer(cols = all_of(elements), names_to = "element") %>%
+      dplyr::select(c(.data[[year]], .data[[month]], .data[[day]], .data$element, .data$value)) %>%
+      dplyr::mutate(value = ifelse(is.na(.data$value), "M", "X")) %>% 
+      tidyr::pivot_wider(id_cols = c(.data$element, .data[[year]], .data[[month]]),
+                         names_from = .data[[day]],
+                         values_from = .data$value) %>%
+      dplyr::rowwise() %>%
+      dplyr::mutate(Available = sum(c(`1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, `10`, `11`, `12`, `13`, `14`, `15`,
+                                      `16`, `17`, `18`, `19`, `20`, `21`, `22`, `23`, `24`, `25`, `26`, `27`, `28`,
+                                      `29`, `30`, `31`) == "X", na.rm = TRUE)) %>%
+      dplyr::mutate(Missing = sum(c(`1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, `10`, `11`, `12`, `13`, `14`, `15`,
+                                    `16`, `17`, `18`, `19`, `20`, `21`, `22`, `23`, `24`, `25`, `26`, `27`, `28`,
+                                    `29`, `30`, `31`) == "M", na.rm = TRUE)) 
+  } else {
+    inventory_data <- data %>%
+        tidyr::pivot_longer(cols = all_of(elements), names_to = "element") %>%
+        dplyr::select(c(.data[[year]], .data[[month]], .data[[day]], .data[[station]], .data$element, .data$value)) %>%
+      dplyr::mutate(value = ifelse(is.na(.data$value), "M", "X")) %>% 
+      tidyr::pivot_wider(id_cols = c(.data[[station]], .data$element, .data[[year]], .data[[month]]),
+                         names_from = .data[[day]],
+                         values_from = .data$value) %>%
+      dplyr::rowwise() %>%
+      dplyr::mutate(Available = sum(c(`1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, `10`, `11`, `12`, `13`, `14`, `15`,
+                                      `16`, `17`, `18`, `19`, `20`, `21`, `22`, `23`, `24`, `25`, `26`, `27`, `28`,
+                                      `29`, `30`, `31`) == "X", na.rm = TRUE)) %>%
+      dplyr::mutate(Missing = sum(c(`1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, `10`, `11`, `12`, `13`, `14`, `15`,
+                                    `16`, `17`, `18`, `19`, `20`, `21`, `22`, `23`, `24`, `25`, `26`, `27`, `28`,
```

Comment:
Actually, ignore my previous comment on this: you could do this with `dplyr::summarise` before you unstack. So you get a separate data frame with the monthly summaries and then merge it back into the unstacked data.